.github/workflows/drepecated-functions-checker.yml

```diff
@@ -0,0 +1,32 @@
+name: "Check Deprecations"
+
+on:
+  pull_request:
+    # Runs on PRs to main
+    branches:
+      - main
+
+jobs:
+  deprecations-check:
+    runs-on: ubuntu-latest
```

Comment:
🤔 We have been advised that self-hosted compute is preferred for workflows in private repos (I am thinking of this workflow running in ENT). 

Personally, I would leave this as `ubuntu-latest` for now, but the self-hosted runners are cheaper and we might need to eventually make this execute on a different type of runner when invoked from ENT.